store/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,380 @@
+	if err := s.batchWriteItems(deleteRequests, s.multihashMapTable); err != nil {
+		return fmt.Errorf("failed to batch delete items: %w", err)
+	}
+
+	return nil
+}
+
+// RemoveProvider implements indexer.Interface.RemoveProvider
+func (s *ddbStore) RemoveProvider(ctx context.Context, providerID peer.ID) error {
+	// Collect items to delete
+	var lastEvaluatedKey map[string]types.AttributeValue
+	var deleteRequests []types.WriteRequest
+
+	for {
+		// Query the providers table for all items with the given providerID
+		result, err := s.client.Query(ctx, &dynamodb.QueryInput{
+			TableName:              aws.String(s.providersTable),
+			KeyConditionExpression: aws.String(fieldProviderID + " = :providerID"),
+			ExpressionAttributeValues: map[string]types.AttributeValue{
+				":providerID": &types.AttributeValueMemberS{Value: providerID.String()},
+			},
+			ExclusiveStartKey: lastEvaluatedKey,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to query provider items for deletion: %w", err)
+		}
+
+		// Process items in the current page
+		for _, item := range result.Items {
+			deleteRequests = append(deleteRequests, types.WriteRequest{
+				DeleteRequest: &types.DeleteRequest{
+					Key: map[string]types.AttributeValue{
+						fieldProviderID: item[fieldProviderID],
+						fieldContextID:  item[fieldContextID],
+					},
+				},
+			})
+		}
+
+		// If there are no more items, we're done
+		if result.LastEvaluatedKey == nil {
+			break
+		}
+		lastEvaluatedKey = result.LastEvaluatedKey
+	}
+
+	// Process all delete requests in batches
+	if err := s.batchWriteItems(deleteRequests, s.providersTable); err != nil {
+		return fmt.Errorf("failed to batch delete provider items: %w", err)
+	}
+
+	return nil
+}
+
+// RemoveProviderContext implements indexer.Interface.RemoveProviderContext
+func (s *ddbStore) RemoveProviderContext(providerID peer.ID, contextID []byte) error {
+	// Create delete item input
+	input := &dynamodb.DeleteItemInput{
+		TableName: aws.String(s.providersTable),
+		Key: map[string]types.AttributeValue{
+			fieldProviderID: &types.AttributeValueMemberS{Value: providerID.String()},
+			fieldContextID:  &types.AttributeValueMemberB{Value: contextID},
+		},
+	}
+
+	// Execute the delete operation
+	_, err := s.client.DeleteItem(context.TODO(), input)
+	if err != nil {
+		// If the item doesn't exist, it's not an error
+		var notFoundErr *types.ResourceNotFoundException
+		if !errors.As(err, &notFoundErr) {
+			return fmt.Errorf("failed to delete provider context: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Size implements indexer.Interface.Size
+func (s *ddbStore) Size() (int64, error) {
+	return 0, nil
+}
+
+// Flush implements indexer.Interface.Flush
+func (s *ddbStore) Flush() error {
+	return nil
+}
+
+// Close implements indexer.Interface.Close
+func (s *ddbStore) Close() error {
+	return nil
+}
+
+// Iter implements indexer.Interface.Iter
+func (s *ddbStore) Iter() (indexer.Iterator, error) {
```

Comment:
Where is this used and will it be a problem we don't support it?

Reply:
there are no references to this method in the codebase